Rocket Generator Problem

**Describe the bug**
I updated generator and created new '@totalsoft/graphql Rocket' server. 
And I've got some errors after "npm start".

**To Reproduce**
Steps to reproduce the behavior:
1. Go to Yeoman menu
2. Click on '@totalsoft/graphql Rocket'
3. Select 'Knex JS' and leave others as default
4. Add DB credentials into .env file
5. Finally run 'npm start' See errors

Here are the errors:

..\graphql-server\src\db\dbConfigService.js
  42:3   error  'connectionInfo' is not defined  no-undef
  44:3   error  'connectionInfo' is not defined  no-undef
  44:43  error  'connectionInfo' is not defined  no-undef
  45:39  error  'connectionInfo' is not defined  no-undef

..\graphql-server\new-server\src\index.js
   23:51  error  "apollo-server-core" is extraneous  node/no-extraneous-require
   67:11  error  'ignore' is not defined             no-undef
   67:76  error  'publicRoute' is not defined        no-undef
  113:9   error  'msgHost' is not defined            no-undef
  115:3   error  'defaultTracer' is not defined      no-undef

..\graphql-server\src\middleware\db\contextDbInstance.js
  2:9   warning  'tenantContextAccessor' is assigned a value but never used  no-unused-vars
  2:43  error    "@totalsoft/multitenancy-core" is not found                 node/no-missing-require

..\graphql-server\src\startup\schema.js
  2:7  warning  'merge' is assigned a value but never used  no-unused-vars

✖ 12 problems (10 errors, 2 warnings)

**Expected behaviour**
After 'npm start' server should be run at 4000 port as expected.

--- a/generators/app/index.js
+++ b/generators/app/index.js
@@ -66,7 +66,7 @@ module.exports = class extends Generator {
     const templatePath = this.templatePath('infrastructure/**/*')
     const destinationPath = this.destinationPath()
 
-    this.ignoreFiles = ['**/.npmignore', '**/.gitignore-template', '**/helm/**']
+    this.ignoreFiles = ['**/.npmignore', '**/.gitignore-template', '**/helm/gql/**']
 
     if (dataLayer === 'knex') this.ignoreFiles = concat(['**/prisma/**'], this.ignoreFiles)
     if (dataLayer === 'prisma')
@@ -135,6 +135,8 @@ module.exports = class extends Generator {
     const packageManagerVersion =
       packageManager === 'npm' ? NPM_MIN_VERSION : packageManager === 'yarn' ? YARN_MIN_VERSION : NPM_MIN_VERSION
     const packageManagerLockFile = packageManager === 'yarn' ? 'yarn.lock' : 'package-lock.json'
+
+    if (!addHelm) this.ignoreFiles = concat(['**/helm/**'], this.ignoreFiles)
 
     this.fs.copyTpl(
       templatePath,

--- a/generators/app/index.js
+++ b/generators/app/index.js
@@ -15,6 +15,7 @@ module.exports = class extends Generator {
   constructor(args, opts) {
     super(args, { ...opts, skipRegenerate: true, ignoreWhitespace: true, force: true, skipLocalCache: false })
     this.registerClientTransforms()
+    this.ignoreFiles = []
   }
 
   async prompting() {
@@ -65,11 +66,11 @@ module.exports = class extends Generator {
     const templatePath = this.templatePath('infrastructure/**/*')
     const destinationPath = this.destinationPath()
 
-    let ignoreFiles = ['**/.npmignore', '**/.gitignore-template', '**/helm/**']
+    this.ignoreFiles = ['**/.npmignore', '**/.gitignore-template', '**/helm/**']
 
-    if (dataLayer === 'knex') ignoreFiles = concat(['**/prisma/**'], ignoreFiles)
+    if (dataLayer === 'knex') this.ignoreFiles = concat(['**/prisma/**'], this.ignoreFiles)
     if (dataLayer === 'prisma')
-      ignoreFiles = concat(
+      this.ignoreFiles = concat(
         [
           '**/src/db/**',
           '**/middleware/db/**',
@@ -81,21 +82,15 @@ module.exports = class extends Generator {
           '**/tracing/knexTracer.js',
           '**/utils/sqlDataSource.js'
         ],
-        ignoreFiles
+        this.ignoreFiles
       )
     if (dataLayer === 'knex' && withMultiTenancy)
-      ignoreFiles = concat(
-        [
-          '**/db/dbInstanceFactory.js',
-          '**/startup/middleware'
-        ],
-        ignoreFiles
-      )
-    if (!addSubscriptions) ignoreFiles = concat(['**/pubSub/**', '**/subscriptions/**'], ignoreFiles)
-    if (!addMessaging) ignoreFiles = append('**/messaging/**', ignoreFiles)
+      this.ignoreFiles = concat(['**/db/dbInstanceFactory.js', '**/startup/middleware'], this.ignoreFiles)
+    if (!addSubscriptions) this.ignoreFiles = concat(['**/pubSub/**', '**/subscriptions/**'], this.ignoreFiles)
+    if (!addMessaging) this.ignoreFiles = append('**/messaging/**', this.ignoreFiles)
 
     if (!withMultiTenancy)
-      ignoreFiles = concat(
+      this.ignoreFiles = concat(
         [
           '**/features/tenant/**',
           '**/multiTenancy/**',
@@ -104,19 +99,28 @@ module.exports = class extends Generator {
           '**/prisma/tenancyFilter.js',
           '**/pubSub/middleware/tenantPublish.js'
         ],
-        ignoreFiles
+        this.ignoreFiles
       )
     if (!hasSharedDb)
-      ignoreFiles = concat(['**/db/multiTenancy/tenancyFilter.js', '**/prisma/tenancyFilter.js'], ignoreFiles)
-    if (!addTracing) ignoreFiles = concat(['**/tracing/**', '**/startup/middleware/tracing.js', '**/pubSub/middlware/tracingPublish.js', '**/__mocks__/opentracing.js'], ignoreFiles)
+      this.ignoreFiles = concat(['**/db/multiTenancy/tenancyFilter.js', '**/prisma/tenancyFilter.js'], this.ignoreFiles)
+    if (!addTracing)
+      this.ignoreFiles = concat(
+        [
+          '**/tracing/**',
+          '**/startup/middleware/tracing.js',
+          '**/pubSub/middlware/tracingPublish.js',
+          '**/__mocks__/opentracing.js'
+        ],
+        this.ignoreFiles
+      )
     if (!withRights)
-      ignoreFiles = concat(
+      this.ignoreFiles = concat(
         ['**/middleware/permissions/**', '**/constants/permissions.js', '**/constants/identityUserRoles.js'],
-        ignoreFiles
+        this.ignoreFiles
       )
 
     if (!addQuickStart)
-      ignoreFiles = concat(
+      this.ignoreFiles = concat(
         [
           '**/features/common/dbGenerators.js',
           '**/features/tenant/**',
@@ -125,7 +129,7 @@ module.exports = class extends Generator {
           '**/middleware/permissions/__tests__/**',
           '**/README.md'
         ],
-        ignoreFiles
+        this.ignoreFiles
       )
 
     const packageManagerVersion =
@@ -137,7 +141,7 @@ module.exports = class extends Generator {
       destinationPath,
       { ...this.answers, packageManagerVersion, packageManagerLockFile },
       {},
-      { globOptions: { ignore: ignoreFiles, dot: true } }
+      { globOptions: { ignore: this.ignoreFiles, dot: true } }
     )
 
     const gitignorePath = this.templatePath('infrastructure/.gitignore-template')
@@ -155,6 +159,11 @@ module.exports = class extends Generator {
         { globOptions: { dot: true } }
       )
     }
+  }
+
+  conflicts() {
+    if (!this.isLatest) return
+    this.ignoreFiles.map(f => this.deleteDestination(f, { globOptions: { onlyFiles: false, dot: true, silent: true } }))
   }
 
   install() {

--- a/generators/app/index.js
+++ b/generators/app/index.js
@@ -132,11 +132,11 @@ module.exports = class extends Generator {
         this.ignoreFiles
       )
 
+    if (!addHelm) this.ignoreFiles = concat(['**/helm/**'], this.ignoreFiles)
+
     const packageManagerVersion =
       packageManager === 'npm' ? NPM_MIN_VERSION : packageManager === 'yarn' ? YARN_MIN_VERSION : NPM_MIN_VERSION
     const packageManagerLockFile = packageManager === 'yarn' ? 'yarn.lock' : 'package-lock.json'
-
-    if (!addHelm) this.ignoreFiles = concat(['**/helm/**'], this.ignoreFiles)
 
     this.fs.copyTpl(
       templatePath,

--- a/generators/app/templates/infrastructure/package.json
+++ b/generators/app/templates/infrastructure/package.json
@@ -36,6 +36,7 @@
     "@graphql-tools/schema": "^8.5.0",
     "@koa/cors": "^3.3.0",
     "@koa/router": "^11.0.1",
+    "koa-ignore": "^1.0.1",
     <%_ if(dataLayer === "prisma") {_%>
     "@prisma/client": "^4.1.1",
     "prisma": "^4.1.1",
@@ -53,7 +54,6 @@
     "@totalsoft/pino-mssqlserver": "^1.1.0",
     <%_ if(withMultiTenancy) {_%>
     "object-path": "^0.11.8",
-    "koa-ignore": "^1.0.1",
       <%_ if(addQuickStart) {_%>
     "datasource-sql": "^2.0.1",
       <%_}_%>
@@ -76,6 +76,7 @@
     "apollo-datasource": "^3.3.2",
     "apollo-datasource-rest": "^3.6.1",
     "apollo-server": "^3.10.0",
+    "apollo-server-core": "^3.10.0",
     "apollo-server-koa": "^3.10.0",
     "async-mutex": "^0.3.2",
     "bluebird": "3.7.2",
@@ -84,7 +85,6 @@
     "dotenv": "16.0.1",
     "graphql": "15.8.0",
     <%_ if(addSubscriptions){ _%>
-    "apollo-server-core": "^3.10.0",
     "graphql-redis-subscriptions": "^2.5.0",
     "graphql-subscriptions": "^2.0.0",
     "graphql-ws": "^5.9.1",
@@ -109,7 +109,6 @@
     "koa": "^2.13.4",
     "koa-bodyparser": "^4.3.0",
     "koa-jwt": "^4.0.3",
-    "lodash.merge": "^4.6.2",
     "mime-types": "^2.1.35",
     "numeral": "^2.0.6",
     "path": "^0.12.7",

--- a/generators/app/templates/infrastructure/src/db/dbConfigService.js
+++ b/generators/app/templates/infrastructure/src/db/dbConfigService.js
@@ -41,8 +41,8 @@ const generateKnexConfig = ({
 })
 
 const getDbConfig = <% if(withMultiTenancy){ %>async ( tenantId )<%} else { %>()<%}%> => {
-  <%_ if(withMultiTenancy){ _%>
   let connectionInfo
+  <%_ if(withMultiTenancy){ _%>
 
   if (isMultiTenant) {
     connectionInfo = await tenantConfiguration.getConnectionInfo(tenantId, '<%= dbConnectionName %>')

--- a/generators/app/templates/infrastructure/src/index.js
+++ b/generators/app/templates/infrastructure/src/index.js
@@ -63,12 +63,11 @@ const
   metrics = require("./monitoring/metrics"),
   metricsPlugin = require("./plugins/metrics/metricsPlugin")
 
-
+  const { publicRoute } = require('./utils/functions'),
+  ignore = require('koa-ignore')
 <%_ if(withMultiTenancy){ _%>
 // MultiTenancy
-const { publicRoute } = require('./utils/functions'),
-  ignore = require('koa-ignore'),
-  { tenantService<% if(dataLayer == "knex") {%>, tenantContextAccessor<%}%> } = require("@totalsoft/multitenancy-core"),
+  ,{ tenantService<% if(dataLayer == "knex") {%>, tenantContextAccessor<%}%> } = require("@totalsoft/multitenancy-core"),
   isMultiTenant = JSON.parse(process.env.IS_MULTITENANT || 'false')
 <%_}_%>
 
@@ -291,9 +290,13 @@ msgHost
 
 async function cleanup() {
   await configMonitor?.close();
+  <%_ if(addMessaging) {_%>
   await msgHost?.stop();
+  <%_}_%>
   await apolloServer?.stop();
+  <%_ if(addTracing) {_%>
   defaultTracer?.close();
+  <%_}_%>
 }
 
 const { gracefulShutdown } = require('@totalsoft/graceful-shutdown');

--- a/generators/app/templates/infrastructure/src/middleware/db/contextDbInstance.js
+++ b/generators/app/templates/infrastructure/src/middleware/db/contextDbInstance.js
@@ -1,5 +1,7 @@
 const { dbInstanceFactory } = require('../../db')
+<%_ if(withMultiTenancy){ _%>
 const { tenantContextAccessor } = require('@totalsoft/multitenancy-core')
+<%_}_%>
 
 const contextDbInstance = () => async (ctx, next) => {
   <%_ if(withMultiTenancy){ _%>

--- a/generators/app/templates/infrastructure/src/startup/schema.js
+++ b/generators/app/templates/infrastructure/src/startup/schema.js
@@ -1,5 +1,4 @@
 const { makeExecutableSchema } = require('@graphql-tools/schema')
-const merge = require('lodash.merge');
 
 const { loadTypedefsSync } = require('@graphql-tools/load'),
   { loadFilesSync } = require('@graphql-tools/load-files'),

